test-framework/common/src/main/java/io/quarkus/test/common/DefaultJarLauncher.java

```diff
@@ -65,23 +114,17 @@ public void start() throws IOException {
         }
         args.add("-jar");
         args.add(jarPath.toAbsolutePath().toString());
+        args.addAll(Arrays.asList(programArgs));
 
         System.out.println("Executing \"" + String.join(" ", args) + "\"");
 
         Files.deleteIfExists(logFile);
         Files.createDirectories(logFile.getParent());
 
-        Function<IntegrationTestStartedNotifier.Context, IntegrationTestStartedNotifier.Result> startedFunction = createStartedFunction();
-
-        quarkusProcess = LauncherUtil.launchProcess(args);
-        if (startedFunction != null) {
```

Comment:
This feels much cleaner to me. I like the function being handled later / by the caller, rather than here.